components/Searchbar.js

```diff
@@ -0,0 +1,151 @@
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useRef, useState, useEffect } from 'react';
+
+export default function Searchbar({ books }) {
+	// Hooks for search bar
+	const searchRef = useRef(null);
+
+	// router
+	const router = useRouter();
+		
+	/**
+	 * @param {HTMLElement} child 
+	 * @param {HTMLElement} parent 
+	 */
+	/*
+	const isDeepChildOf = (child, parent) => {
```

Comment:
This is commented out. Is it because it will potentially be used later?

Reply:
Yes - it's part of a feature I couldn't get working, maybe for another task. I will add a comment on this